src/app/components/authmodel/AuthModel.js

```diff
@@ -0,0 +1,226 @@
+    signup: <RegisterForm setAuthStates={setAuthStates} username={username} setUsername={setUsername} />, 
+    confirm:<ConfirmSignUpForm setAuthStates={setAuthStates} username={username}/>
+}
+
+  return (
+    <div className="w-screen z-30 h-screen fixed inset-0 bg-gray-800 bg-opacity-50 flex items-center justify-center">
+      <div className="bg-white p-6 rounded-lg shadow-lg w-96 relative z-50">
+        <button className="absolute top-2 right-2 text-gray-500 hover:text-gray-700"
+          onClick={() => setShowModal(false)} >X</button>
+        {obj[authStates]}
+
+        <button
+          className="mt-4 text-blue-500 hover:text-blue-700"
+          onClick={() => setAuthStates(authStates==='login'?'signup':'login')}
+        >
+          {authStates==='login' ? 'Switch to Register' : authStates==='signup' ? 'Switch to Login' : ''}
+          
+        </button>
+      </div>
+    </div>
+  )
+}
+
+const LoginForm = ({setShowModal, setAuthStates, username, setUsername, setIsSignin}) => {
+
+  const [password, setPassword] = useState('')
+  const [message, setMessage] = useState('')
+
+  const handleLogin = async (e) => {
+    e.preventDefault()
+    setMessage('')
+    try {
+      const { isSignedIn, nextStep } = await signIn({ username, password })
+      console.log(isSignedIn, nextStep)
+      console.log((nextStep['signInStep']))
+      setMessage(nextStep.signInStep)
+        switch(nextStep.signInStep){
+          case 'CONFIRM_SIGN_IN_WITH_SMS_MFA_CODE':
+            break
+            // TODO: Handle MFA if app supports it
+          case 'CONFIRM_SIGN_IN_WITH_NEW_PASSWORD':
+            break
+            // TODO: prompt user to enter a new password
+          case 'RESET_PASSWORD':
+            break
+            // TODO: prompt user to reset their password
+          case 'CONFIRM_SIGN_UP':
+            setMessage('该账户未验证，请检查您的电子邮件以验证账户。')
+            setTimeout(() => {setAuthStates('confirm')}, 800)
+            break
+            // TODO: prompt user to confirm sign up 
+          case 'DONE':
+            setMessage('登录成功')
+            setTimeout(() => {setIsSignin(true);setShowModal(false)}, 800)
+            break
+        default:
+            break
+        }
+
+     
+    } catch (error) {
+      setMessage(error.message)
+    }
+  }
+
+  return (
+    <div>
+      <h2 className="text-2xl mb-4 text-slate-700">Login</h2>
+      <form onSubmit={handleLogin}>
+        <input
+          type="email"
+          placeholder="Email"
+          className="w-full p-2 mb-4 border rounded dark:text-slate-900"
+          value={username}
+          onChange={(e) => setUsername(e.target.value)}
+        />
+        <input
+          type="password"
+          placeholder="Password"
+          className="w-full p-2 mb-4 border rounded dark:text-slate-900"
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
```

Comment:
Same for autocomplete and credentials saving purpose, it's recommended to use `autoComplete='password'`, [ref](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete#values)